### PR TITLE
Add MPLAB X IDE

### PR DESCRIPTION
**Pull request by CyberCastle**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [X] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/mplabx-ide.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+cask 'mplabx-ide' do
+  version '5.40'
+  sha256 '00f582ff074845260c7d2474c92a99b0f0f1cb4e16a7e6d173c5cabd27da0fa3'
+
+  url "https://ww1.microchip.com/downloads/en/DeviceDoc/MPLABX-v#{version}-osx-installer.dmg"
+  appcast 'https://macupdater.net/cgi-bin/check_urls/check_url_redirect.cgi?url=https://www.microchip.com/mplabx-ide-osx-installer'
+  name 'MPLab X IDE'
+  homepage 'https://www.microchip.com/mplab/mplab-x-ide'
+
+  app 'mplab_ide.app', target: 'microchip/mplab_ide.app'
```
Comment (vitorgalvao) [inline Casks/mplabx-ide.rb:10]: Why the rename?

Comment (CyberCastle) [reply Casks/mplabx-ide.rb:10]: As such, it is not a rename, it is an ordering of the applications, just as the original installer does. I used [this](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/kicad.rb) barrel as a reference.

File: Casks/mplabx-ide.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                                    '--othermcu', '0',
+                                    '--exepermission', 'no',
+                                    '--collectInfo', '0',
+                                    '--installdir', staged_path.to_s
+                                  ],
+                      input:      ['y'],
+                      sudo:       true,
+                    }
+
+  postflight do
```
Comment (vitorgalvao) [inline Casks/mplabx-ide.rb:28]: Do you need this ownership change to use the software?

File: Casks/mplabx-ide.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    set_ownership '/Applications/microchip'
+  end
+
+  uninstall script: {
+                      executable: "Uninstall_MPLAB_X_IDE_v#{version}.app/Contents/MacOS/installbuilder.sh",
+                      args:       ['--mode', 'unattended'],
+                      input:      ['y', 3],
+                      sudo:       true,
+                    },
+            delete: ['/Applications/microchip']
```
Comment (vitorgalvao) [inline Casks/mplabx-ide.rb:39]: Use `#{appdir}`, not `/Applications`

File: Casks/mplabx-ide.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/mplabx-ide.rb:10]: > just as the original installer does.

What exactly does this mean?

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/mplabx-ide.rb (hunk 1 above)
Comment (CyberCastle) [reply Casks/mplabx-ide.rb:10]: The installer adds other application in "/Applications/microchip" folder, by default, and this cannot be changed. So use that folder to install mplab_ide.app and mplab_ipe.app. That is why the configuration of this cask has been done in this way.

File: Casks/mplabx-ide.rb (hunk 2 above)
Comment (CyberCastle) [reply Casks/mplabx-ide.rb:28]: Yes, especially if it is required to adjust some settings or to uninstall it.

File: Casks/mplabx-ide.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/mplabx-ide.rb:10]: Thank you. That clarifies it.

File: Casks/mplabx-ide.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                                    '--collectInfo', '0',
+                                    '--installdir', staged_path.to_s
+                                  ],
+                      input:      ['y'],
+                      sudo:       true,
+                    }
+
+  postflight do
+    set_ownership staged_path.to_s
+    set_ownership "#{appdir}/microchip"
```
Comment (vitorgalvao) [inline Casks/mplabx-ide.rb:30]: > The installer adds other application in "/Applications/microchip" folder, by default, and this cannot be changed.

That makes my request for changing this incorrect.

**Review by vitorgalvao — APPROVED** [review]

**Comment by CyberCastle** [comment]
@vitorgalvao Thanks you for merge this cask. Can you check this other [cask](https://github.com/Homebrew/homebrew-cask/pull/83106) please?

**Pull request merged**